staging/src/k8s.io/api/core/v1/generated.proto

```diff
@@ -3170,6 +3170,16 @@ message PodStatus {
   // +optional
   optional string reason = 4;
 
+  // nominatedNodeName is set only when this pod preempts other pods on the node, but it cannot be
+  // scheduled right away as preemption victims receive their graceful termination periods.
+  // This field does not guarantee that the pod will be scheduled on this node. Scheduler may decide
+  // to place the pod elsewhere if other nodes become available sooner. Scheduler may also decide to
+  // give the resources on this node to a higher priority pod that is created after preemption.
+  // As a result, this field may be different than PodSpec.nodeName when the pod is
```

Comment:
Why is it desirable to allow them to be different? Why not either set them to match or clear the nominatedNodeName once nodeName is set?

Reply:
There are a couple of reasons:
1. It is useful to understand/debug behavior of the cluster. When a pod causes preemption, but it is scheduled elsewhere, checking nominatedNodeName reveals that this is the pod that caused  preemption on the node. Users who don't have access to scheduler logs can still see this field.
2. Performance: setting the nominatedNodeName to match the nodeName or clearing it requires another API call.